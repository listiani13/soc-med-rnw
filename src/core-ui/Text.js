// @flow

import React from 'react';
import {Text as TextComp, StyleSheet} from 'react-native';
import {FONT_SIZE} from '../constants/text';
import {BAHAMA_BLUE} from '../constants/color';

type Props = {
  size?: 'xxsmall' | 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge',
  weight?: 'light' | 'reg' | 'bold' | 'black',
  primary?: boolean,
  color?: string,
  style?: StyleSet,
  fontWeight?: '100' | '200' | '300' | '400' | '500' | 'bold',
  children: ReactNode | string,
};

function Text(props: Props) {
  let {
    primary,
    weight,
    style,
    color,
    size,
    children,
    fontWeight,
    ...otherProps
  } = props;
  let fontSize = FONT_SIZE[size || 'medium'];
  let fontStyle = [
    styles.default,
    {fontSize, fontWeight},
    {color},
    primary && styles.primary,
    style && style,
  ];
  return (
    <TextComp style={fontStyle} {...otherProps}>
      {children}
    </TextComp>
  );
}

export default Text;

const styles = StyleSheet.create({
  default: {
    fontSize: FONT_SIZE['medium'],
    fontWeight: '300',
    color: BAHAMA_BLUE,
  },
  primary: {
    color: BAHAMA_BLUE,
  },
});
