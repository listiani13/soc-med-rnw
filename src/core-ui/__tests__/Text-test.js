// @flow
import React from 'react';
import {render} from 'react-testing-library';
import Text from '../Text';

describe('Text component', () => {
  it('should render small text', () => {
    let {container} = render(<Text size="small">Hello</Text>);
    expect(container).toMatchSnapshot();
  });
  it('should render bold large text', () => {
    let {container} = render(
      <Text size="large" fontWeight="500">
        Hello
      </Text>,
    );
    expect(container).toMatchSnapshot();
  });
});
