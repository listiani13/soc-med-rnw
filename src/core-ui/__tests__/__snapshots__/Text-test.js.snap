// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render bold large text 1`] = `
<div>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
    style="font-size: 28px; font-weight: 500;"
  >
    Hello
  </div>
</div>
`;

exports[`Text component should render small text 1`] = `
<div>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
    style="font-size: 15px;"
  >
    Hello
  </div>
</div>
`;
